features/onboarding/passkey/useLoginWithPasskey.ts

```diff
@@ -0,0 +1,70 @@
+import { useCallback, useEffect } from "react";
+import { usePasskeyAuthStoreContext } from "./passkeyAuthStore";
+import { usePrivy } from "@privy-io/expo";
+import { useLoginWithPasskey as useLoginWithPasskeyPrivy } from "@privy-io/expo/passkey";
+import { usePrivyAuthStoreContext } from "../Privy/privyAuthStore";
+import { captureErrorWithToast } from "@/utils/capture-error";
+import { RELYING_PARTY } from "./passkey.constants";
+
+/**
+ * This hook is used to login with an existing Passkey account and set the account id in the Passkey Auth Store
+ * It is expected to be used with usePrivySmartWalletConnection, but does not include that hook as this will likely be used with login with passkey
+ * And we would run into issues with effects with how Privy is setup
+ * @returns {
+ *  loginWithPasskey: () => Promise<void>
+ * }
+ */
+export const useLoginWithPasskey = () => {
+  // Privy Hooks
+  const { logout, user: privyUser } = usePrivy();
+  const { loginWithPasskey, state: loginState } = useLoginWithPasskeyPrivy();
+
+  // Passkey Store Hooks
+  const { setLoading, setStatusString, setError } = usePasskeyAuthStoreContext(
+    (state) => state
+  );
+
+  // Privy Auth Store Hooks
+  const setPrivyAccountId = usePrivyAuthStoreContext(
+    (state) => state.setPrivyAccountId
+  );
+
+  useEffect(() => {
+    setStatusString(loginState.status);
+  }, [setStatusString, loginState.status]);
+
+  const handleLoginWithPasskey = useCallback(async () => {
+    try {
+      setLoading(true);
+      if (privyUser) {
+        // We support multiple accounts, so we need to logout of the current before creating a new one
```

Comment:
Same thing here we should throw instead of implicitly doing anything and require the collar to have logged out prior to calling this method